### PR TITLE
java resources

### LINKED ISSUE #703: I want to add Java resources under language section and we can add even more languages
### Description

THere are currently 4 languages in this section, so I am willing to add the section of Java as well as Nodejs, so it will be even more helpful if all language resources are in one place, and that is the only aim of LINKSHUB.  So, please Assign this to me! @rupali-codes 

### Screenshots

_No response_

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by P766-spec**
Hello @rupali-codes , I have done with adding java resources from which I particularly learned java, not just resources added youtube channels also! Closes #703
So if you found it a valuable contribution, please accept it.
Thank you!

### CODE REVIEW

**Comment by rupali-codes** [comment]
@P766-spec hey could you please reference the issue correctly? please add _closes #703_ in the description of your PR

**Comment by P766-spec** [comment]
Done!

**Comment by P766-spec** [comment]
I have done nothing with .vscode , could you please specify what exactly I got to do??

**Comment by P766-spec** [comment]
Okay got with .vscode wala , Donw with uncommit
now could you please review it and let me know if there is any other thing? @rupali-codes

File: database/languages/java.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        "name": "Java Fundamentals by javatpoint",
+        "description": "This site teaches fundamentals of the Java programming language in an amazing and  simplified way",
+        "url": "https://www.javatpoint.com/java-tutorial",
+        "category": "Languages",
+        "subcategory": "java"
+    },
+    {
+        "name": "Bro Code",
+        "description": "This playlist contains all the fundamental concepts that every Java learner should know, Bro code teaches it in a simple manner with visualization",
```
Comment (CBID2) [inline database/languages/java.json:11]: [Proofreading]
```suggestion
        "description": "This playlist contains all the fundamental concepts that every Java learner should know. Bro code teaches these concepts in a simple manner with visuals",
```

File: database/languages/java.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        "url": "https://youtu.be/NBIUbTddde4",
+        "category": "languages",
+        "subcategory": "java"
+    },
+    {
+        "name": "DSA with TutorialsPoint",
+        "description": "This site provides all basic as well some advanced topic of DSA using Java with examples",
```
Comment (CBID2) [inline database/languages/java.json:18]: [Proofreading]
```suggestion
        "description": "This site provides all basic and advanced lessons on DSA and Java with examples",
```

File: database/languages/java.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        "url": "https://www.tutorialspoint.com/dsa_using_java/index.htm",
+        "category": "languages",
+        "subcategory": "java"
+    },
+    {
+        "name": "Kunal Kushwaha",
+        "description": "This playlist covers the DSA concepts with good number of problems as well as porblems to solve DSA by yuor own",
```
Comment (CBID2) [inline database/languages/java.json:25]: [Typos]
```suggestion
        "description": "This playlist covers the DSA concepts with good number of problems as well as DSA exercises you solve on your own",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @P766-spec! :) I found some typos and proofreading errors in your descriptions. In the future, consider proofreading your descriptions with a tool like Grammarly before submitting them as a pull request. :)

**Comment by P766-spec** [comment]
Ouhhh okayyy Sorry!, will do it @rupali-codes @CBID2

**Comment by P766-spec** [comment]
Done with the suggested changes! @CBID2

**Review by CBID2 — APPROVED** [review]
LGTM

**Review by rupali-codes — CHANGES_REQUESTED** [review]
@P766-spec please uncommit your changes under .vscode dir, other than that, looks good to me

**Comment by P766-spec** [comment]
> @P766-spec please uncommit your changes under .vscode dir, other than that, looks good to me

I have done it already ?? didn't understood what exactly your are talking about
Screenshot (617)

**Comment by rupali-codes** [comment]
> > @P766-spec please uncommit your changes under .vscode dir, other than that, looks good to me
> 
> I have done it already ?? didn't understood what exactly your are talking about Screenshot (617)

I can still see that in _filechanges_. I'm asking you to remove that from the PR 😊

**Review by ujjawaltyagii — COMMENTED** [review]
@P766-spec I don't think we have `Java` sub-category in `languages` section. So you basically just added bunch of resources to the database with no frontend to show those resources !

1. Try creating a new sub-category first if any category doesn't exist
2. Run and test things on your local system to see if your changes are being reflected or not and ADD screenshots in the PR description as it will save a lot of time while reviewing PR :)

**Comment by P766-spec** [comment]
Okayyy sure will do it

**Comment by P766-spec** [comment]
> 

I am Sorry but could you please tell me how to remove it?

**Comment by rupali-codes** [comment]
@P766-spec checkout [this](https://gist.github.com/gunjanpatel/18f9e4d1eb609597c50c2118e416e6a6)

**Comment by P766-spec** [comment]
> @P766-spec checkout [this](https://gist.github.com/gunjanpatel/18f9e4d1eb609597c50c2118e416e6a6)

Screenshot (621)
what shall I do now I am new to this revert one please tell

File: .vscode/settings.json [hunk 4]
```diff
@@ -11,4 +11,5 @@
   "[json]": {
     "editor.defaultFormatter": "vscode.json-language-features"
   }
+
```
Comment (rupali-codes) [inline .vscode/settings.json:14]:
```suggestion
```

Comment (P766-spec) [reply .vscode/settings.json:14]: ok

**Pull request merged**